### DISCUSSION TITLE
Move pandas as a core requirement?

### DISCUSSION THREAD

**Discussion by mwtoews** [modflowpy/flopy#1788] [General]
Currently pandas is an optional dependency ([described here](https://github.com/modflowpy/flopy/blob/develop/docs/flopy_method_dependencies.md)).

@w-bonelli noted [here](https://github.com/modflowpy/flopy/pull/1783#discussion_r1194505695) that it should be moved as a core dependency, since it is almost as common as NumPy, based on GH stats.

I wouldn't have suggested this idea a few years ago (when pandas was version < 1), but I think pandas is now mature and as wide-spread as numpy. If pandas is moved into a core dependency, it could replace the use of NumPy recarray structures over time.

Any other pros? What are the downsides of adding pandas as a required dependency?

**Comment by christianlangevin** [comment]
I think it's time to add pandas as a core dependency.  It seems like some projects were burned when pandas moved from 0.x to 1.x and then to 2.x, but pandas seems much more stable than it was when a few years ago.  We have some ongoing work to more closely integrate pandas with the underlying mf6 flopy data structures in order to substantially improve read/write speeds.

**Comment by wpbonelli** [comment]
@jdhughes-usgs this was discussed in the call today and had group support, @langevin-usgs suggested checking if you had any concerns and going ahead with it if not

**Comment by mwtoews** [comment]
Another question is to the timing. If it is agreed to bring pandas as a core requirement, which release? Before or after the next release?

**Comment by wpbonelli** [comment]
pandas moved into the core with #1887
